test/base/test_helpers.py

```diff
@@ -388,6 +388,46 @@ def test_convert_uptime_string_seconds(self):
         self.assertEqual(convert_uptime_string_seconds('95w2d10h58m'), 57668280)
         self.assertEqual(convert_uptime_string_seconds('1h5m'), 3900)
 
+    def test_canonical_interface_name(self):
+        """Test the canonical_interface_name helper function."""
+        self.assertEqual(napalm.base.helpers.canonical_interface_name('Fa0/1'), "FastEthernet0/1")
+        self.assertEqual(napalm.base.helpers.canonical_interface_name('FastEthernet0/1'),
+                         'FastEthernet0/1')
+        self.assertEqual(napalm.base.helpers.canonical_interface_name('TenGig1/1/1.5'),
+                         "TenGigabitEthernet1/1/1.5")
+        self.assertEqual(napalm.base.helpers.canonical_interface_name('Gi1/2'),
+                         "GigabitEthernet1/2")
+        self.assertEqual(napalm.base.helpers.canonical_interface_name('HundredGigE105/1/1'),
+                         "HundredGigabitEthernet105/1/1")
+        self.assertEqual(napalm.base.helpers.canonical_interface_name('Lo0'), "Loopback0")
+        self.assertEqual(napalm.base.helpers.canonical_interface_name('lo0'), "Loopback0")
+        self.assertEqual(napalm.base.helpers.canonical_interface_name('no_match0/1'),
+                         "no_match0/1")
+        self.assertEqual(napalm.base.helpers.canonical_interface_name('lo',
+                         addl_name_map={"lo": "something_custom"}), "something_custom")
+        self.assertEqual(napalm.base.helpers.canonical_interface_name('uniq',
```

Comment:
Same should be `uniq0` and return `something_custom0`. I think this is more consistent with usage pattern.